client/driver/spawn/spawn_test.go

```diff
@@ -5,64 +5,81 @@ import (
 	"io/ioutil"
 	"os"
 	"os/exec"
-	"runtime"
 	"strings"
 	"testing"
 	"time"
 )
 
-func TestSpawn_NoCmd(t *testing.T) {
-	f, err := ioutil.TempFile("", "")
-	if err != nil {
-		t.Fatalf("TempFile() failed")
+func TestMain(m *testing.M) {
+	switch os.Getenv("TEST_MAIN") {
+	case "app":
+		appMain()
+	default:
+		os.Exit(m.Run())
+	}
+}
+
+func appMain() {
+	if len(os.Args) < 2 {
+		fmt.Fprintln(os.Stderr, "no command provided")
+		os.Exit(1)
+	}
+	switch cmd := os.Args[1]; cmd {
+	case "echo":
+		fmt.Println(strings.Join(os.Args[2:], " "))
+	case "sleep":
```

Comment:
Nice. I was trying to figure out how to do this on Windows and I found a bunch of hacks, but this one should be fairly portable in our case.